lib/graphql/sugar/object.rb

```diff
@@ -1,21 +1,37 @@
 module GraphQL
   module Sugar
     module Object
+      COMMON_ATTRIBUTES = [:id, :created_at, :updated_at]
+
       def self.included(base)
         base.extend ClassMethods
       end
 
       module ClassMethods
+        def inherited(klass)
+          derived_model_class = klass.name.demodulize.gsub('Type', '').safe_constantize
+          klass.instance_variable_set(:@model_class, derived_model_class)
+          klass.instance_variable_set(:@attributes_initialized, Set.new)
+        end
+
         def model_class(model_class)
           @model_class = model_class
+          initialize_common_attributes unless all_initial_attributes_added?
+        end
 
-          common_attribute_names = [:id, :created_at, :updated_at]
-          common_attribute_names.each(&method(:attribute))
+        def initialize_common_attributes
+          COMMON_ATTRIBUTES.each do |_attribute|
+            @attributes_initialized.add(_attribute)
```

Comment:
Remove `@attributes_initialized.add(_attribute)` from within `initialize_common_attributes` as that it is being handled by `attribute` method.

Reply:
I need to add the attribute to the `@attributes_initialized` Set, so that it won't be called again from the `attribute` method.